ledger/benches/blockstore_processor.rs

```diff
@@ -0,0 +1,176 @@
+        bank.store_account(
+            &account.pubkey(),
+            &Account {
+                lamports: 5100,
+                data: vec![],
+                owner: system_program::id(),
+                executable: false,
+                rent_epoch: Epoch::MAX,
+            },
+        );
+    });
+
+    accounts
+}
+
+fn create_transactions(bank: &Bank, num: usize) -> Vec<SanitizedTransaction> {
+    let funded_accounts = create_funded_accounts(bank, 2 * num);
+    funded_accounts
+        .into_par_iter()
+        .chunks(2)
+        .map(|chunk| {
+            let from = &chunk[0];
+            let to = &chunk[1];
+            system_transaction::transfer(from, &to.pubkey(), 1, bank.last_blockhash())
+        })
+        .map(SanitizedTransaction::from_transaction_for_tests)
+        .collect()
+}
+
+struct BenchFrame {
+    bank: Arc<Bank>,
+    prioritization_fee_cache: PrioritizationFeeCache,
+}
+
+fn setup(apply_cost_tracker_during_replay: bool) -> BenchFrame {
+    let mint_total = u64::MAX;
+    let GenesisConfigInfo {
+        mut genesis_config, ..
+    } = create_genesis_config(mint_total);
+
+    // Set a high ticks_per_slot so we don't run out of ticks
+    // during the benchmark
+    genesis_config.ticks_per_slot = 10_000;
+
+    let mut bank = Bank::new_for_benches(&genesis_config);
+
+    if !apply_cost_tracker_during_replay {
+        bank.deactivate_feature(&apply_cost_tracker_during_replay::id());
+    }
+
+    // Allow arbitrary transaction processing time for the purposes of this bench
+    bank.ns_per_slot = u128::MAX;
+
+    // set cost tracker limits to MAX so it will not filter out TXs
+    bank.write_cost_tracker()
+        .unwrap()
+        .set_limits(std::u64::MAX, std::u64::MAX, std::u64::MAX);
+    let bank = bank.wrap_with_bank_forks_for_tests().0;
+    let prioritization_fee_cache = PrioritizationFeeCache::default();
+    BenchFrame {
+        bank,
+        prioritization_fee_cache,
+    }
+}
+
+fn bench_execute_batch(
+    bencher: &mut Bencher,
+    batch_size: usize,
+    apply_cost_tracker_during_replay: bool,
+) {
+    let BenchFrame {
+        bank,
+        prioritization_fee_cache,
+    } = setup(apply_cost_tracker_during_replay);
+    let transactions = create_transactions(&bank, 2_usize.pow(20));
+    let batches: Vec<_> = transactions
+        .chunks(batch_size)
+        .map(|txs| {
+            let mut batch =
+                TransactionBatch::new(vec![Ok(()); txs.len()], &bank, Cow::Borrowed(txs));
+            batch.set_needs_unlock(false);
+            TransactionBatchWithIndexes {
+                batch,
+                transaction_indexes: (0..batch_size).collect(),
+            }
+        })
+        .collect();
+    let mut batches_iter = batches.into_iter();
+
+    let mut timing = ExecuteTimings::default();
+    bencher.iter({
+        let bank = bank.clone();
+        move || {
+            let batch = batches_iter.next().unwrap();
+            execute_batch(
```

Comment:
I'd prefer to process the same total number of transactions regardless `batch_size` to show the overhead more clearly:

```patch
diff --git a/ledger/benches/blockstore_processor.rs b/ledger/benches/blockstore_processor.rs
index e0d19853fe..7ec2b17d97 100644
--- a/ledger/benches/blockstore_processor.rs
+++ b/ledger/benches/blockstore_processor.rs
@@ -130,16 +130,18 @@ fn bench_execute_batch(
 
     let mut timing = ExecuteTimings::default();
     bencher.iter(|| {
-        let batch = batches_iter.next().unwrap();
-        execute_batch(
-            &batch,
-            &bank,
-            None,
-            None,
-            &mut timing,
-            None,
-            &prioritization_fee_cache,
-        )
+        for _ in 0..(64/batch_size) {
+            let batch = batches_iter.next().unwrap();
+            execute_batch(
+                &batch,
+                &bank,
+                None,
+                None,
+                &mut timing,
+                None,
+                &prioritization_fee_cache,
+            ).unwrap();
+        }
     });
 }

```

result:

```
running 6 tests
test bench_execute_batch_full_batch                        ... bench:     740,169 ns/iter (+/- 39,800)
test bench_execute_batch_full_batch_disable_tx_cost_update ... bench:     774,346 ns/iter (+/- 28,495)
test bench_execute_batch_half_batch                        ... bench:     824,189 ns/iter (+/- 29,661)
test bench_execute_batch_half_batch_disable_tx_cost_update ... bench:     811,608 ns/iter (+/- 20,936)
test bench_execute_batch_unbatched                         ... bench:   1,381,782 ns/iter (+/- 49,475)
test bench_execute_batch_unbatched_disable_tx_cost_update  ... bench:   1,334,157 ns/iter (+/- 88,541)

test result: ok. 0 passed; 0 failed; 0 ignored; 6 measured; 0 filtered out; finished in 42.13s

```

Reply:
Yeah that's a great idea!

Reply:
c0d5b46

Reply:
Didn't use `take` since that will consume the iterator and not let us use it the next iteration of the benchmark.

Reply:
I made a similar change for the consumer benchmarks: #34752

Thanks for the suggestion, no more math to compare the throughput